### ISSUE TITLE
Misleading error when using fully qualified trait syntax without trait

### ISSUE THREAD

**Issue by Cerber-Ursi** [rust-lang/rust#98565]
Given the following code: 

```rust
trait StringExt {
    fn ext(&self);
}
impl StringExt for String {
    fn ext(&self) {
        <Self as str>::trim(self);
        // str::trim(self); - this works
    }
}
```
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f3ca5eea2c043a3bdf681d6c3e864b67)

The current output is:

```
error[E0576]: cannot find method or associated constant `trim` in `str`
 --> src/lib.rs:6:24
  |
6 |         <Self as str>::trim(self);
  |                        ^^^^ not found in `str`

For more information about this error, try `rustc --explain E0576`.
```

However, `str` is not a trait, therefore the fully-qualified trait syntax should be simply invalid in this case - therefore, the error would better say the real problem ("`str` is not a trait"), probably with the help message suggesting the commented-out line.

Initially found in [this SO question](https://stackoverflow.com/questions/72766027/cannot-find-method-or-associated-constant-join-in-stdthreadjoinhandle).

**Comment by lyming2007** [comment]
Did you mean the ideal output should like:
```
error[E0576]: cannot find method or associated constant `trim` in `str`
 --> src/lib.rs:6:24
  |
6 |         <Self as str>::trim(self);
  |                        ^^^^ `str` is not a trait

For more information about this error, try `rustc --explain E0576`.
```
?

**Comment by Cerber-Ursi** [comment]
Not sure, to be honest. This might event not be E0576, since the problem is not that "something wasn't found for the type" - it's "there's no type to search in".

**Comment by estebank** [comment]
Current output:

```
error[E0404]: expected trait, found builtin type `str`
 --> src/lib.rs:6:18
  |
6 |         <Self as str>::trim(self);
  |                  ^^^ not a trait
```

**Comment by jannden** [comment]
I would like to give it a try and add the test for this.

@rustbot claim
